### PR TITLE
Update client for v1.12

### PR DESCRIPTION
**Pull request by coszio**
Updates the client with the latest proto definitions, by following the contributing guide. Everything went smoothly 🏄 

- Adds sanity tests for the new facet and distance matrix endpoints
- No new OneOf's to add manual helpers. The only new ones are only used for response types.

### CODE REVIEW

**Review by Anush008 — APPROVED** [review]
LGTM 🙏🙏

File: qdrant_test/points_test.go [hunk 1]
```diff
@@ -182,6 +192,39 @@ func TestPointsClient(t *testing.T) {
...
+		sample := uint64(10)
+		limit := uint64(10)
+		res, err := client.SearchMatrixOffsets(ctx, &qdrant.SearchMatrixPoints{
+			CollectionName: collectionName,
+			Sample:         &sample,
+			Limit:          &limit,
+		})
+		require.NoError(t, err)
+		require.Empty(t, res)
+	})
```
Comment (coszio) [inline qdrant_test/points_test.go:226]: Example usage of the new endpoints

cc @agourlay

**Comment by coszio** [comment]
Example usage of the new endpoints:

### SearchMatrixOffsets
```go
import (
	"context"

	"github.com/qdrant/go-client/qdrant"
)

client, err := qdrant.NewClient(&qdrant.Config{
	Host: "localhost",
	Port: 6334,
})

sample := uint64(10)
limit := uint64(10)
res, err := client.SearchMatrixOffsets(ctx, &qdrant.SearchMatrixPoints{
	CollectionName: "{collection_name}",
	Sample:         &sample,
	Limit:          &limit,
})
```

### SearchMatrixPairs
```go
import (
	"context"

	"github.com/qdrant/go-client/qdrant"
)

client, err := qdrant.NewClient(&qdrant.Config{
	Host: "localhost",
	Port: 6334,
})

sample := uint64(10)
limit := uint64(10)
res, err := client.SearchMatrixPairs(ctx, &qdrant.SearchMatrixPoints{
	CollectionName: "{collection_name}",
	Sample:         &sample,
	Limit:          &limit,
})
```

### Facet
```go
import (
	"context"

	"github.com/qdrant/go-client/qdrant"
)

client, err := qdrant.NewClient(&qdrant.Config{
	Host: "localhost",
	Port: 6334,
})

sample := uint64(10)
limit := uint64(10)
res, err := client.Facet(ctx, &qdrant.FacetCounts{
	CollectionName: "{collection_name}",
	Key:            "key",
})
```

**Comment by generall** [comment]
don't seem to be used in facet
```
sample := uint64(10)
limit := uint64(10)
```

**Comment by coszio** [comment]
Sorry, too much copy paste

**Pull request merged**